tensorrt_llm/_torch/modules/fused_moe/create_moe.py

```diff
@@ -36,6 +39,17 @@ def get_moe_cls(
                 f"Check out details in quant_config: {quant_config}"
                 "Using CutlassFusedMoE instead.")
             return CutlassFusedMoE
+    elif moe_backend.upper() == "WIDEEP":
+        all_to_all_method_type = WideEPMoE.select_alltoall_method_type(
+            model_config.mapping, routing_method.experts_per_token, dtype,
+            model_config.use_cuda_graph)
+        if all_to_all_method_type != AlltoallMethodType.NotEnabled:
+            return WideEPMoE
+        else:
+            logger.warning(
+                "WideEPMoE only supports MNNVL or DeepEP. Check out WideEPMoE.select_alltoall_method_type "
+                "for more details. Using CutlassFusedMoE instead.")
+            return CutlassFusedMoE
```

Comment:
It's confusing if the user specifies `moe_backend.upper() == "WIDEEP"` but we create `CutlassFusedMoE` because `mapping.moe_ep_size <= top_k` (in `select_alltoall_method_type`). It's not a problem before this PR because there is no "WIDEEP" choice.

Futher, I don't think `select_alltoall_method_type` can be called multiple times for one layer. If `MNNVL` is determined on class selection, we should use the decision in `WideEPMoE` rather than call `select_alltoall_method_type` again.

Reply:
How about that I change it to 
```
elif moe_backend.upper() == "WIDEEP":
    return WideEPMoE
```
You guys can keep refining it as you wish.

Reply:
👍

Reply:
Updated.